Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -229,15 +226,46 @@ class EdgeHitProcessor: HitProcessing {
         return requestHeaders
     }
 
+    private func getRequestProperties(from event: Event) -> [String: Any]? {
+        var requestProperties = [String: Any]()
+        if let overwritePath = getCustomRequestPath(from: event), !overwritePath.isEmpty {
+            requestProperties[EdgeConstants.EventDataKeys.Request.PATH] = overwritePath
+        }
+        return requestProperties
+    }
+
     private func getCustomRequestPath(from event: Event) -> String? {
-        var customPath: String?
+        var path: String?
         if let eventData = event.data {
-            let xdmData = eventData[EdgeConstants.JsonKeys.XDM] as? [String: Any]
-            let requestData = xdmData?[EdgeConstants.JsonKeys.REQUEST] as? [String: Any]
-            customPath = requestData?[EdgeConstants.JsonKeys.PATH] as? String
+            let requestData = eventData[EdgeConstants.EventDataKeys.Request.KEY] as? [String: Any]
+            path = requestData?[EdgeConstants.EventDataKeys.Request.PATH] as? String
+        }
+
+        guard let customPath = path else {
+            return nil
+        }
+
+        if !isValidPath(customPath) || customPath.contains("//") {
+            Log.error(label: self.SELF_TAG, "Dropping the overwrite path value: (\(String(describing: customPath))), since it contains invalid characters.")
+            return nil
         }
 
         return customPath
     }
 
+    private func isValidPath(_ path: String?) -> Bool {
+        guard let path = path else {
+            return false
+        }
+
+        let pattern = "^\\/[/.a-zA-Z0-9-]+$"
```

Comment:
Let's allow all the unreserved characters in a URL, so add underscore `_` and tilde `~`.
Unreserved characters in URL include upper and lower Alpha, Numeric, plus `.`, `_`, `-`, and `~`.
See https://www.ietf.org/rfc/rfc3986.txt section 2.3